R/financial_functions.R

```diff
@@ -0,0 +1,28 @@
+#' Calculates annual net profits on the company-technology level for the
+#' baseline and late and sudden scenarios. Climate laggards which need to build
+#' out their production in increasing technologies to compensate for their
+#' missed targets, are "punished" by adjusting the net profit margin on their
+#' additional build out based on their proximity to target and market share within
+#' the given technology. This ensures that late build out will not
+#' proportionally translate into increased profits. The adjustment for the
+#' market share of the company accounts for the fact that companies of different
+#' size will face different degrees of difficulty catching up.
+#'
+#' @param data A data frame containing the production forecasts of companies
+#'   under baseline and late and sudden, market prices/costs, company net profit
+#'   margins, the proximity to target in the production forecast period and an
+#'   indication of the direction of the technology.
+calculate_net_profits <- function(data) {
+  data <- data %>%
+    dplyr::mutate(
+      production_compensation = .data$late_sudden - .data$baseline,
+      net_profits_baseline = .data$baseline * .data$Baseline_price * .data$net_profit_margin,
+      net_profits_ls = dplyr::if_else(
+        .data$direction == "declining",
```

Comment:
just playing devils advocate, why is it not neccesary to also do some proximity to target-dependent handling for increasing technologies? Just asking because this looks assymetric.

Reply:
The contrary is happening. If the technology is `declining`, the basic profit function is used. if it is not `declining` it will get penalised based on the proximity.
This is not happening for decreasing technologies, because a compensation for decreasing technologies is a punishment on the profits per construction. It means less production for laggards, hence less profits.
For increasing, the inverse is the case. Compensation means additional production for laggards. What this punishment is doing is to say that this additional production cannot simply be passed to the same profit function because that would reflect unrealistic and perverse incentives. Not building out such technologies and then being forced to increase them would lead to a sudden need for strong investments and more costs though, which is what we are approximating here